src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -2281,7 +2283,12 @@ def process_gallery_image_version_namespace(cmd, namespace):
                         "usage error: {} is an invalid target region cvm encryption. "
                         "Both os_cvm_encryption_type and os_cvm_des parameters are required.".format(cvm_terms))
 
-                storage_profile_types = [profile_type.value for profile_type in ConfidentialVMEncryptionType]
+                storage_profile_types = [
+                    "EncryptedVMGuestStateOnlyWithPmk",
+                    "EncryptedWithPmk",
+                    "EncryptedWithCmk",
+                    "NonPersistedTPM"
+                ]
```

Comment:
Do we need to hard code the valid enumeration values? Could they be obtained from the data model of Code Gen?

Reply:
There's no data model for Code Gen. In Code Gen, enumerable parameters are defined with hard coded enums like:
![image](https://github.com/user-attachments/assets/8d925706-4c98-4257-8b94-07bf465529be)

Reply:
Got it, thanks~